### PR TITLE
PKMN: Make  Exp All early

### PR DESCRIPTION
**Pull request by Joethepic**
Please format your title with what portion of the project this pull request is
targeting and what it's changing.

ex. "MyGame4: implement new game" or "Docs: add new guide for customizing MyGame3"

## What is this fixing or adding?
title

## How was this tested?
2 test generations with m64, mc, alttp, cf, and 2 pokemon red and blue games one with the setting off and one with it on.
Then verifying if Exp All was early when it should be and not early when it shouldn't be.

### CODE REVIEW

File: worlds/pokemon_rb/options.py [hunk 1]
```diff
@@ -25,7 +25,6 @@ class RivalName(TextChoice):
     option_choose_in_game = -1
     default = -1
 
-
```
[the remark this answers is not in the document]
Comment (Joethepic) [reply worlds/pokemon_rb/options.py:28]: fixed

**Comment by Joethepic** [comment]
@Alchav this probably needs your attention

**Comment by Alchav** [comment]
I don't like that this would lead two there being "early oak's parcel" and "early exp all" options but which work differently, as the early exp all setting does not seem to force the item to be a local item while early oak's parcel does. Furthermore, if someone wants EXP all that badly they can just put it in start_inventory, which I often do

**Comment by Joethepic** [comment]
The reason I didn’t force it to be local is because on early parcel or vanilla there would be no or not many sphere 1 locations in world to place it and if playing on open varidian city can just make it in local_items.

**Comment by Berserker66** [comment]
> I don't like that this would lead two there being "early oak's parcel" and "early exp all" options but which work differently, as the early exp all setting does not seem to force the item to be a local item while early oak's parcel does. Furthermore, if someone wants EXP all that badly they can just put it in start_inventory, which I often do

So want me to close this PR?

**Comment by Alchav** [comment]
Tell you what, I'd accept having it always make it an early item and remove the option

**Comment by Joethepic** [comment]
Will change that tonight then

**Review by Alchav — APPROVED** [review]

File: worlds/pokemon_rb/options.py [hunk 2]
```diff
@@ -506,6 +506,7 @@ class LoseMoneyOnBlackout(Toggle):
     default = 1
 
 
+
```
Comment (Berserker66) [inline worlds/pokemon_rb/options.py:509]: this newline breaks pep8

File: worlds/pokemon_rb/__init__.py [hunk 3]
```diff
@@ -171,6 +171,9 @@ def encode_name(name, t):
         # damage being reduced by 1 which leads to a "not very effective" message appearing due to my changes
         # to the way effectiveness messages are generated.
         self.type_chart = sorted(chart, key=lambda matchup: -matchup[2])
+        self.multiworld.early_items[self.player]["Exp. All"] = 1
+ 
+
```
Comment (Berserker66) [inline worlds/pokemon_rb/__init__.py:176]: these newlines break pep8

**Comment by Joethepic** [comment]
I’ll look into those review fixes tomorrow

**Review by Berserker66 — APPROVED** [review]

**Pull request merged**